src/java/org/apache/cassandra/utils/FBUtilities.java

```diff
@@ -939,6 +939,33 @@ public static String prettyPrintMemory(long size, String separator)
                                  UNIT_PREFIXES.charAt(UNIT_PREFIXES_BASE + prefixIndex));
     }
 
+    /**
+     * Convert the given size in bytes to a human-readable value using binary (i.e. 2^10-based) modifiers
+     * with only three significant digits.
+     * For example, "4.96 KiB", "48.8 KiB", "477 MiB", "0.422 KiB", "0.000 KiB".
+     * @param size      Number to convert.
+     */
+    public static String prettyPrintMemoryShort(long size)
```

Comment:
can you just walk us through this method conceptually? It is a little bit hard to follow to be honest.

Reply:
Flow chart of the above:
```mermaid
flowchart TD
    A[Calculate Prefix Index] -->|Determine prefix index based on size| B[Adjust for Smallest Prefix]
    B -->|Set to 'Kilo' for sizes < 1024 bytes| C[Get Prefix Character]
    C -->|Retrieve unit prefix character| D[Scale the Size]
    D -->|Divide size by 2^ 10×prefixIndex | E[Format Scaled Size]
    E -->|Is scaled size between -1 and 1?| F[Format Small Numbers]
    E -->|Format Larger Numbers| G[Format Large Numbers]
    F & G --> H[Merge Scaled Size with Prefix]
    H -->|Construct final human-readable string| I[End]
```

Reply:
A playground on replit that implements this logic and tests it: https://replit.com/@tytakoff/CASSANDRA-19104#Main.java

Reply:
@zaaath can't we just use ` org.apache.cassandra.io.util.FileUtils.stringifyFileSize`?

Reply:
@smiklosovic thanks for the code pointer, I like the output that `org.apache.cassandra.io.util.FileUtils.stringifyFileSize`  produces, but please note that it doesn't quite match what was agreed on in the Jira and in the mailing list:
1. "KiB might make sense as the lowest reportable units."
2. "With a leading zero before the decimal point, there should be three significant digits afterwards"

Comparison:
```
Expected output and FBUtilities.prettyPrintMemoryShort:
0: 0.000 KiB
5: 0.005 KiB
50: 0.049 KiB
500: 0.488 KiB
5000: 4.88 KiB
50000: 48.8 KiB
500000: 488 KiB
5000000: 4.77 MiB
50000000: 47.7 MiB
500000000: 477 MiB

FileUtils.stringifyFileSize:
0: 0 bytes
5: 5 bytes
50: 50 bytes
500: 500 bytes
5000: 4.88 KiB
50000: 48.83 KiB
500000: 488.28 KiB
5000000: 4.77 MiB
50000000: 47.68 MiB
500000000: 476.84 MiB
```

I'll switch the logic to `FileUtils.stringifyFileSize` if you think it's warranted to not fully follow the expected output format. I'm on board with this.

Reply:
@smiklosovic I migrated the code to `org.apache.cassandra.io.util.FileUtils.stringifyFileSize` as you suggested. Let me know how it looks.